py-rattler/Cargo.toml

```diff
@@ -48,7 +49,7 @@ tokio = { version = "1.32" }
 thiserror = "1.0.44"
 url = "2.4.1"
 
-openssl = { version = "0.10", optional = true }
+openssl = { version = "0.10", optional = true, features = ["vendored"] }
```

Comment:
We have features in this crate that control this behavior. See rustls-tls, nativetls and openssl-vendored.

Reply:
So I added this just as an attempt to fix the issues I'm seeing in the CI related to the openssl-sys crate. Do you have any idea what's going on? I ran this successfully on a linux machine I have access to, and when I uninstalled libssl-dev then I got a similar error message, but not sure how my PR could have caused this. Possibly the latest version of openssl-sys changed? Curious if you've seen this before

Reply:
There are two libraries that provide TLS for our networking operations (on linux). `openssl` (through `native-tls`) and `rustls`. We default to use `native-tls` (so `openssl` on linux) but allow using `rustls`. We use `rustls` for the single executable binaries of `pixi` but use `openssl` when building for conda-forge.

The problem is that if one rattler crate depends on another rattler crate you need to propagate the proper tls features otherwise the "default" features "leak" through the compilation.

In CI we add the proper features. When we build the wheels for py-rattler we [add the `vendored-openssl` feature](https://github.com/mamba-org/rattler/blob/66cad536862c28bae09f80efd8fe29442822194b/.github/workflows/release-python.yml#L95). 

So, in essence, you can remove this change and use features to get the desired behavior for different targets.